spring-integration-cassandra/src/main/java/org/springframework/integration/cassandra/config/xml/CassandraOutboundChannelAdapterParser.java

```diff
@@ -0,0 +1,114 @@
+
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.config.RuntimeBeanReference;
+import org.springframework.beans.factory.support.AbstractBeanDefinition;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.ManagedMap;
+import org.springframework.beans.factory.xml.ParserContext;
+import org.springframework.integration.cassandra.outbound.CassandraMessageHandler;
+import org.springframework.integration.cassandra.outbound.CassandraMessageHandler.OperationType;
+import org.springframework.integration.config.ExpressionFactoryBean;
+import org.springframework.integration.config.xml.AbstractOutboundChannelAdapterParser;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.xml.DomUtils;
+import org.w3c.dom.Element;
+
+/**
+ * @author Filippo Balicchia
+ *
+ */
+public class CassandraOutboundChannelAdapterParser extends AbstractOutboundChannelAdapterParser
+{
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected AbstractBeanDefinition parseConsumer(Element element, ParserContext parserContext)
+    {
+        BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(CassandraMessageHandler.class);
+        
+        
+        String cassandraTemplate = element.getAttribute("cassandra-template");
+        String operationType = element.getAttribute("operation-type");
+        String writeOptionsInstance = element.getAttribute("write-options");
+        String ingestQuery = element.getAttribute("cql-ingest");
+        String outputChannel = element.getAttribute("output-channel");
+        String query = element.getAttribute("query");
+        String producesReply = element.getAttribute("produces-reply");
+
+		if (StringUtils.isEmpty(cassandraTemplate)){
+			parserContext.getReaderContext().error("cassandra-template is empty", element);
+		}
+
+		if (StringUtils.isEmpty(operationType)) {
+			parserContext.getReaderContext().error("operation-type need to be specified", element);
+		}
+		
+		OperationType queryType = CassandraMessageHandler.OperationType.toType(operationType);
+		builder.addConstructorArgValue(new RuntimeBeanReference(cassandraTemplate));
+		builder.addPropertyValue("queryType", queryType);
+		
+		if(StringUtils.isNotEmpty(writeOptionsInstance)){
+			builder.addPropertyReference("writeOptions", writeOptionsInstance);
+		}
+		
+		
+		if(StringUtils.isNotEmpty(outputChannel)){
+			builder.addPropertyReference("outputChannel", outputChannel);
+		}
+		
+		if (StringUtils.isNotEmpty(ingestQuery) && !"INSERT".equalsIgnoreCase(operationType)) {
+			parserContext.getReaderContext().error( "Ingest cql query can be apply only with insert operation",element);
+		}
+		else if (StringUtils.isNotEmpty(ingestQuery)) {
+			builder.addPropertyValue("ingestQuery", ingestQuery);
+		}
+		
+        if (StringUtils.isNotEmpty(query)){
```

Comment:
You still have some issues with formatting, so I'm forced to comment on the matter :smile: 
Minor, of course, but you should be ready do not recognize your code in the `master` after my polishing on merge.
E.g. the `{` should be on the same line as its context and with the whitespace before it.